### PR TITLE
Support RSK

### PR DESCRIPTION
**Pull request by fvictorio**
This PR does some modifications that allow the bridge UI to work with RSK:

- Don't use events to check if the transaction was processed, since RSK's public node doesn't support the `getLogs` RPC method.
- Handle transaction receipt status differently (see [this comment](https://github.com/poanetwork/bridge-ui/pull/138#discussion_r223095345))
- Don't show an error when events cannot be fetched. Since the stores are always updating the last events (for using them in the events section), we cannot avoid using them.

### CODE REVIEW

**Comment by igorbarinov** [comment]
Deploy preview for *kind-kilby-95344f* processing.

Built with commit bc808005b5ca9320fcec8ec3fe83c3578e0dc2d7

https://app.netlify.com/sites/kind-kilby-95344f/deploys/5bb7cd6f73f2cf3c813484aa

**Comment by akolotov** [comment]
@fvictorio could you merge the changes from `add-support-erc20-native` to your branch as so we could test the bridge with RSK nodes?

**Comment by fvictorio** [comment]
Merged and resolved conflicts.

I tested that everything works in native-to-erc between Sokol and Kovan. I couldn't test it between RSK and Sokol because the bridge address I have (`0x1e16aa4Df73d29C029d94CeDa3e3114EC191E25A`) doesn't seem to have code anymore in the RSK testnet.

**Comment by patitonar** [comment]
@akolotov e2e tests were failing because the success alert after transaction was processed was missing, so I fixed that.

The only thing is that on the success alert we used to display a message with the transaction hash (with the link to explorer) from the event that generated the transfer on the other network.
 
Now that we don't use events to check if the transaction was processed, we don't have a way to get that transaction hash. So for example the message only displays: `Tokens received on Kovan`.

If this is OK then I think all the changes are done for this PR.

If displaying the tx hash on the alert is important, then we can try using both approaches, using events when the network supports `getLogs` methods (in which case we can display the tx hash) and using the new approach on networks as RSK which doesn't support this method. This will add some complexity to the code but it can be done.

**Comment by akolotov** [comment]
@patitonar thanks for looking after this PR.

I agree that we need a parameter in the `.env` file to allow configuration what kind of network is used on one or another side of the bridge. My suggestion is `..._WITHOUT_EVENTS`

These options will also allow to make decision if we need to display elements that depends on events: the event tab, the bridge statistic elements on the statistics tab.

File: public/index.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
         <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
         <meta name="theme-color" content="#000000">
         <meta property="og:title" content="POA Bridge UI" />
-        <meta property="og:description" content="The POA cross-chain bridge serves as a method of transferring POA native tokens from the POA Network to the Ethereum network in a quick and cost-efficient manner. " />
+        <meta property="og:description" content="The TokenBridge serves as a method of transferring the Bancor Network tokens between the Ethereum network to Rootstock network in a quick and cost-efficient manner." />
```
Comment (akolotov) [inline public/index.html]: We need to have a separate parameter for this in the `.env` file

**Comment by fernandomg** [comment]
> @patitonar thanks for looking after this PR.
> 
> I agree that we need a parameter in the `.env` file to allow configuration what kind of network is used on one or another side of the bridge. My suggestion is `..._WITHOUT_EVENTS`
> 
> These options will also allow to make decision if we need to display elements that depends on events: the event tab, the bridge statistic elements on the statistics tab.

Done

File: public/index.html (hunk 1 above)
Comment (fernandomg) [reply public/index.html]: Done

File: .env.example [hunk 2]
```diff
@@ -21,3 +21,9 @@ REACT_APP_FOREIGN_GAS_PRICE_ORACLE_URL=https://gasprice.poa.network/
 REACT_APP_FOREIGN_GAS_PRICE_SPEED_TYPE=standard
 REACT_APP_FOREIGN_GAS_PRICE_FALLBACK=5000000000
 REACT_APP_FOREIGN_GAS_PRICE_UPDATE_INTERVAL=15000
+
+# Default
+REACT_APP_DESCRIPTION="The POA cross-chain bridge serves as a method of transferring POA native tokens from the POA Network to the Ethereum network in a quick and cost-efficient manner."
```
Comment (akolotov) [inline .env.example]: Please update the README file with the new parameter.

File: .env.example [hunk 3]
```diff
@@ -7,6 +7,10 @@ REACT_APP_HOME_NATIVE_NAME=POA
 REACT_APP_HOME_NETWORK_NAME="POA Sokol"
 REACT_APP_FOREIGN_NETWORK_NAME=Kovan
 
+# Uncomment the constant for the network that don't support events
+#REACT_APP_HOME_WITHOUT_EVENTS=1
```
Comment (akolotov) [inline .env.example]: Does it make sense to use "true' instead of "1"?

File: .env.example [hunk 4]
```diff
@@ -7,6 +7,10 @@ REACT_APP_HOME_NATIVE_NAME=POA
...
+#REACT_APP_FOREIGN_WITHOUT_EVENTS=1
```
Comment (akolotov) [inline .env.example]: Please update the README file with new parameters.

Comment (fernandomg) [reply .env.example]: Done

File: .env.example (hunk 3 above)
Comment (fernandomg) [reply .env.example]: Done

File: .env.example (hunk 2 above)
Comment (fernandomg) [reply .env.example]: Done

File: .env.example (hunk 3 above)
Comment (fernandomg) [reply .env.example]: @patitonar, @akolotov changed. Now by default both constants are `false`, and the user will need to changed them to `true` to inform of a network _without events_

**Comment by fernandomg** [comment]
> Tokens Deposits/Withdraws sections also uses Events to get that information. So I think the Statistics tab should also not be displayed if we can't use events.
Hidden from the menu for non-event based networks.

> List of authorities is generated by listening to events ValidatorAdded and ValidatorRemoved. So it shouldn't be displayed if we can't use events.
Done

> On configuration section. The Authorities count is calculated from the previously mentioned list of validators. In this case I think we can use validatorCount method from BridgeValidators contract to get this information.
Done

Also, implemented _redirect_ for the urls that are not supported in a non-event based network. So if the user tries to access: `/events` it will be redirected to `/`; same for `/statistics`

@patitonar

**Comment by akolotov** [comment]
Do you have ideas why the deployment checks failed?

**Comment by fernandomg** [comment]
> Do you have ideas why the deployment checks failed?

Yes, `yn` (library to parse `true`/`false` values for `.env` values) is incompatible with the version of `react-script` that we're using in this project.

**Comment by akolotov** [comment]
@patitonar if you have no more comments please approve the PR and I will merge it. Thanks!

**Pull request merged**